[justincred] iP

justincred opened the pull request:
Jux
>“There are not many things in life worth missing **sleep** for” – Justin Lim Chong Hwei written at 5.10am 😴 

Jux frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://github.com/justincred/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 - [X] Managing tasks
 - [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

justincred committed:
Rename, Greet, Exit

justincred committed:
echoing

justincred committed:
list

justincred committed:
mark as done

justincred committed:
mark as done part 2

justincred committed:
Todos,Events, Deadline - inheritance (unpolished)

justincred committed:
testing p1

justincred committed:
Added some errors

justincred committed:
Refactoring some code, added delete fn, added more errors with extra for future

justincred committed:
added saving to hard disk feature

justincred committed:
Merge branch 'branch-level-7'

linnn-7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Task> list = new ArrayList<>(100);
```
Maybe can use the plural form for this variable name?

justincred replied:
thanks for pointing it out!

linnn-7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
+                String input = scanner.nextLine();
+                if (input.equals("bye")) {
+                    break;
+                } else if(input.equals("list")) {
```
Maybe should add a blank space after "if"?
```suggestion
                } else if (input.equals("list")) {
```

linnn-7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
+                        String listMessage = j + "." +  list.get(i);
+                        System.out.println(listMessage);
+                    }
+                } else if(input.startsWith("mark")) {
```
```suggestion
                } else if (input.startsWith("mark")) {
```

linnn-7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
+                    }
+                } else if(input.startsWith("mark")) {
+
+                    if (input.length() > 5 ) {
```
```suggestion
                    if (input.length() > 5) {
```

linnn-7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,175 @@
+                        int convertedToNumber = Integer.parseInt(listStringNumber) - 1;
+                        // future error when list is empty
+                        if (convertedToNumber < 0 || convertedToNumber >= list.size()) {
+                            throw new DukeException("NUMBER NOT IN LIST, PLEASE ADD A TASK OR " +
+                                    "CHOOSE A DIFFERENT NUMBER WITHIN 1 AND"
+                                    + list.size());
```
I like how you provide users with thorough instructions on how to enter the right data!

linnn-7 commented on src/main/java/FileManipulation.java:
```diff
@@ -0,0 +1,71 @@
+            char isDoneChar = data.charAt(2);
+            boolean isDone = isDoneChar == 1;
+            switch(data.charAt(0)) {
+                case 'T':
+                    list.add(new Todo(data.substring(4), isDone));
+                    break;
+                case 'D':
+                    int dateIndex = Util.findNthDivider(data, 3);
+                    list.add(new Deadline(data.substring(4,dateIndex), isDone, data.substring(dateIndex)));
+                        break;
+                case 'E':
+                    int dateStartIndex = Util.findNthDivider(data, 3);
+                    int dateEndIndex = Util.findNthDivider(data,4);
+                    list.add(new Event(data.substring(4,dateStartIndex), isDone,
+                            data.substring(dateStartIndex, dateEndIndex), data.substring(dateEndIndex)));
+                    break;
+                default:
+                    break;
```
Maybe can update the IDE setting so that there is no indentation for case clauses

justincred replied:
Yep, did that already, thank you

linnn-7 reviewed (commented):
I like how you structure your OOP designs, and I find your code clear and easy to read. Your instructions for users are also well-crafted, and I appreciate the clarity they provide. While I observed a few minor violations of coding standards, overall, I like your project.

justincred committed:
Add Date and Time with unfinished error handling

justincred committed:
Merge branch 'branch-Level-8'

justincred committed:
Fix DateTime format and error handling

justincred committed:
Refactored code for better OOP

justincred committed:
Merge branch 'A-MoreOOP'

justincred committed:
Refactor code and add packages

justincred committed:
Merge branch 'branch-Packages'

justincred committed:
Merge branch 'add-gradle'

justincred committed:
Add gradle

justincred committed:
Add testing for parser task inputs and deadline string format

justincred committed:
Add JavaDocs

justincred committed:
Merge branch 'branch-A-JavaDoc'

justincred committed:
Refactor code for coding standards

justincred committed:
fix merge conflict

justincred committed:
Add find feature

justincred committed:
Fix merge conflict

qinxutan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,286 @@
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+public class Parser {
```
easily understood your code! :) good job

qinxutan commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,128 @@
+
+import java.util.ArrayList;
+import java.util.Scanner;
+public class Ui {
```
well structured ui!

qinxutan reviewed (commented):
LGTM! All the best for the rest of your ip :)

justincred committed:
Add Checkstyle and reinstall gradle

justincred committed:
Merge branch 'branch-A-CheckStyle'

justincred committed:
Implement GUI skeleton with JavaFx

justincred committed:
Partial sync GUI with storage

justincred committed:
Fix bug of task list size not updating

justincred committed:
Fix missing deleted task in ui

justincred committed:
Change the Gui startup and include chatbot history

justincred committed:
Merge branch 'branch-A-Level-10'

justincred committed:
Add assertions

Add assertions to confirm assumptions in several places
*In the various add task methods in tasklist
to confirm the correct number of arguments

justincred committed:
Refactor parser classes for readability

justincred committed:
Further refactor parser class

justincred committed:
Add JavaDocs and more refactoring

justincred committed:
Merge pull request #1 from justincred/branch-A-Assertions

Add assertions

justincred committed:
Merge pull request #2 from justincred/branch-A-CodeQuality

Refine code quality

justincred committed:
Add more date options

Add more date time options for user such as /Tues 17:00 or
/2024-09-09 or /09-09-2024
Also ensure that if day chosen is current day and time is
before current time, the date chosen will be next week instead
Refactor date time parsing to another class for readability

justincred committed:
Merge branch 'branch-BCD-extension'

justincred committed:
Fix input bug of dates using slashes /

justincred committed:
Add Extend Dialogbox

justincred committed:
Refactor code and add more error messages

justincred committed:
Fix DateTimeParser bug:

justincred committed:
Add further testing

justincred committed:
Add name to GUI

justincred committed:
Add User Guide and change user picture

justincred committed:
Change filepath launcher

justincred committed:
Remove other Readme

justincred committed:
Update README.md
